### PR TITLE
feat(firewall): define rate limit for network topics

### PR DESCRIPTION
**Pull request by b00f**
## Description

This PR defines rate limits for network topics. 
The rate limit specifies the number of messages per second that are allowed. 
If the threshold is exceeded, the message won't be broadcasted.
It also define new topics for block and transactions.

### CODE REVIEW

File: network/gossip.go [hunk 1]
```diff
@@ -12,21 +13,23 @@ import (
...
-	generalTopicName string
-	rateLimit        *rateLimit
+	ctx                    context.Context
+	wg                     sync.WaitGroup
+	config                 *Config
+	host                   lp2phost.Host
+	pubsub                 *lp2pps.PubSub
+	topics                 []*lp2pps.Topic
+	subs                   []*lp2pps.Subscription
+	generalTopicDeprecated *lp2pps.Topic
```
Comment (Ja7ad) [inline network/gossip.go:23]: Instead rename topic to deprecated you can use standard comment for deprecated, for example:

```go
generalTopic *lp2pps.Topic // Deprecated: generalTopic ...
```

File: network/gossip.go [hunk 2]
```diff
@@ -58,15 +61,130 @@ func newGossipService(ctx context.Context, host lp2phost.Host, eventCh chan Even
...
+		}
+
+		return g.BroadcastMessage(msg, g.consensusTopic)
+
+	default:
+		return InvalidTopicError{TopicID: topicID}
+	}
+}
+
+func (g *gossipService) JoinGeneralTopic(sp ShouldPropagate) error {
```
Comment (Ja7ad) [inline network/gossip.go:110]: Add comment Deprecated for this method:

```go
// Deprecated: JoinGeneralTopic
func (g *gossipService) JoinGeneralTopic(sp ShouldPropagate) error {}
```

File: network/gossip.go [hunk 3]
```diff
@@ -58,15 +61,130 @@ func newGossipService(ctx context.Context, host lp2phost.Host, eventCh chan Even
...
+	topic, err := g.JoinTopic(g.consensusTopicName(), sp)
+	if err != nil {
+		return err
+	}
+	g.consensusTopic = topic
+
+	return nil
+}
+
+func (g *gossipService) generalTopicNameDeprecated() string {
```
Comment (Ja7ad) [inline network/gossip.go:170]: Please add comment deprecated instead rename method:

```go
// Deprecated: generalTopicName
func (g *gossipService) generalTopicName() string {}
```

File: network/interface.go [hunk 4]
```diff
@@ -9,15 +9,23 @@ import (
 type TopicID int
 
 const (
-	TopicIDGeneral   TopicID = 1
-	TopicIDConsensus TopicID = 2
+	TopicIDGeneralDeprecated TopicID = -1
```
Comment (Ja7ad) [inline network/interface.go:12]: Please add comment deprecated instead rename enum:

```go
TopicIDGeneral   TopicID = -1 // Deprecated: TopicIDGeneral
```

**Review by Ja7ad — CHANGES_REQUESTED** [review]

File: network/gossip.go (hunk 1 above)
Comment (b00f) [reply network/gossip.go:23]: Done.

File: network/gossip.go (hunk 3 above)
Comment (b00f) [reply network/gossip.go:170]: Done.

File: network/interface.go (hunk 4 above)
Comment (b00f) [reply network/interface.go:12]: Done.

**Review by Ja7ad — APPROVED** [review]

**Comment by Ja7ad** [comment]
@b00f Thank you

**Pull request merged**